### DISCUSSION TITLE
Eco Start to take account of dynamic tariffs (e.g. agile)

### DISCUSSION THREAD

**Discussion by Wiggy93** [nathanmarlor/foxess_em#327] [Ideas]
Is it plausible to enable the eco start/end to take sensors tracking a variable rate (e. g. Agile) to input a eco Start and end period dynamically? It could use a sum of those periods to work out if the battery needs to charge during any/all of these cheaper rates.

**Comment by mattwilson1066** [comment]
Has anyone set up anything yet which can pull the data for the costs for the day, and then pick the cheapest X amount of time slots and then pass that into the inverter?

Below are two sources of data for the prices which are usually published between 4pm and 8pm for the next day.
https://agileprices.co.uk/?region=J
https://agile.octopushome.net/dashboard

If this hasn't been looked at yet, would love to work with someone to spec out the requirements etc.

**Reply by Wiggy93** [reply]
The unofficial octopus integration has [target rate sensors](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/blob/develop/_docs/setup_target_rate.md) that you can set up to identify X time slots (either continuous or broken up in chunks) that I've used to build something similar to the octopus compare app agile dashboard. 

These are binary sensors that have a 'target_times' attribute with the list of applicable times you can use to trigger automations. I use this to pull out the next start time of the cheapest 1h continous block for setting washing machine etc going.

`- name: agile_continous_1h`
`   state: >`
`      {{ as_timestamp(state_attr('binary_sensor.octopus_energy_target_agile_continous_1h', 'target_times')[0]['valid_from'], 
            "date_time_iso") | timestamp_custom("$a %H:%M) }}`

No idea how to pass this onto the inverter though and don't have the electrical know how to set up a modbus connection to my inverter to use Nathan's other integration. I only started learning how to use home assistant/yaml 2 weeks ago !

**Comment by mattwilson1066** [comment]
Amazing thank you. I will have a proper read of this tonight.
In terms of the modbus connection, thats actually the easy bit which I did not struggle with. I have not really done any coding or this side of things since college, but the wiring bit I have found easier. Happy to help you with that if needed.
Using a 'usb to rs485 waveshare' HA just picks it up without any hassle, went through the setup steps he shared. Wired the 2 pin outs from the inverter to the USB stick and I am getting real time data.

**Reply by Wiggy93** [reply]
Have you got any links to videos/pictures of what I should expect it to look like?
FYI with the coding, in HA developer -> template, its good for testing the syntax of the code as it can otherwise not be happy!

**Reply by mattwilson1066** [reply]
Let me pull together a few images of the wiring, links to the how to and how I have done mine, and I will post a few this evening.

**Reply by Wiggy93** [reply]
Any luck with getting pics of the wiring etc?

**Reply by mattwilson1066** [reply]
This is for the H1 inverter, for the rs485 its 3 and 4, I used ferrule crimps on them as it makes them much easier to work with, but you don't need to. So that is the green pair. The blue pair are connected to a CT coming from my other inverter which has my solar panels connected to it, this also isn't necessary, the only thing it does is show up the source on the map of where the data is flowing on the fox app. It also means you can use the CT2 data in Home Assistant, as my other inverter is a bit more complex to connect to.
![PXL_20230809_195711806-COLLAGE](https://github.com/nathanmarlor/foxess_em/assets/112546497/483cd42e-71e7-43e9-bf49-100063cfb242)

The image below shows the twisted pair going into the 'usb to rs485 waveshare' as I was able to run a Cat5, but you can use RS485 to ethernet or wifi adapters. 
![PXL_20230815_175937020](https://github.com/nathanmarlor/foxess_em/assets/112546497/c8a792ba-d8ca-4c28-aec9-184c686f8388)
